frontend/src/app/logs/logs.component.ts

```diff
@@ -169,15 +179,28 @@ export class LogsComponent implements OnInit, OnDestroy {
       return;
     }
     const formValue = this.searchForm.value;
-    const criteria: LogSearchParam = {
-      start: formValue.start!,
-      end: formValue.end!,
-      messageContent: formValue.messageContent!,
-      levels: formValue.levels!,
-      scopeTypes: formValue.scopeTypes!,
-      scopeIds: this.selectedScopes!.map(scope => scope.scopeId),
-      page: 0
-    };
+    let criteria: LogSearchParam;
+    if (this.scopeId !== null && this.scopeType !== null) {
+      criteria = {
+        start: formValue.start!,
+        end: formValue.end!,
+        messageContent: formValue.messageContent!,
+        levels: formValue.levels!,
+        scopeTypes: [this.scopeType],
+        scopeIds: [this.scopeId],
+        page: 0
+      };
+    } else {
+      criteria = {
+        start: formValue.start!,
+        end: formValue.end!,
+        messageContent: formValue.messageContent!,
+        levels: formValue.levels!,
+        scopeTypes: formValue.scopeTypes!,
+        scopeIds: this.selectedScopes!.map(scope => scope.scopeId),
+        page: 0
+      };
+    }
```

Comment:
Instead of having this whole if, you can use a ternary condition on scopeTypes and scopeIds fields. Something like : `scopeTypes = this.scopeType ? [this.scopeType] : formValue/scopeTypes!`